R/PackageFunctionReporter.R

```diff
@@ -86,13 +86,25 @@ PackageFunctionReporter <- R6::R6Class(
             }
             
             log_info(sprintf('Constructing network representation...'))
-            funcMap <- mvbutils::foodweb(where = paste("package", private$packageName, sep = ":"), plotting = FALSE)
+            # foodweb will output a warning for "In par(oldpar) : calling par(new=TRUE) with no plot" all the time. 
+            # does not seem to be an issue
+            funcMap <- suppressWarnings(mvbutils::foodweb(where = paste("package"
+                                                                        , private$packageName
+                                                                        , sep = ":")
+                                                          , plotting = FALSE))
+            
             log_info("Done constructing network representation")
             
             # Function Connections: Edges
             edges <- data.table::melt(data.table::as.data.table(funcMap$funmat, keep.rownames = TRUE)
                                       , id.vars = "rn")[value != 0]
+            
+            # Formattign
+            edges[, value := NULL]
             data.table::setnames(edges,c('rn','variable'), c('TARGET','SOURCE'))
+            edges[, c('TARGET','SOURCE') := lapply(.SD, as.character)
```

Comment:
this makes me nervous, since it relies on ordering. I know I'm being super nitpicky and probably worrying about nothing, but given that we're only modifying two columns and we know their names, I don't think we need to deal with the complexities of `.SD`

I recommend just changing this to this:

```
edges[, TARGET := as.character(TARGET)]
edges[, SOURCE := as.character(SOURCE)]
```

IMO the complexity and loss of readability from using `.SDcols` is only worthwhile when you're modifying many columns or when you're dynamically defining a list of column names to be modified